From 750d4e82a0c1161292d24216bcff200cd6647611 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Thu, 29 Jul 2010 11:39:40 -0700
Subject: [PATCH] XQuartz: xpbproxy: Don't take down the whole server on an IO error

Calls pthread_exit to prevent _XIOError from calling exit()

This fixes http://xquartz.macosforge.org/trac/ticket/421

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 hw/xquartz/pbproxy/main.m |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git xorg-server-1.8.99.905/hw/xquartz/pbproxy/main.m xorg-server-1.8.99.905/hw/xquartz/pbproxy/main.m
index 560cf01..bb06465 100644
--- xorg-server-1.8.99.905/hw/xquartz/pbproxy/main.m
+++ xorg-server-1.8.99.905/hw/xquartz/pbproxy/main.m
@@ -75,6 +75,8 @@ static int x_io_error_handler (Display *dpy) {
         exit(EXIT_FAILURE);
 #endif
 
+    /* Prevent _XIOError from calling exit() */
+    pthread_exit(NULL);
     return 0;
 }
 
-- 
1.5.6.6

